src/components/Editor/index.js

```diff
@@ -16,44 +17,199 @@ export default class Editor extends Component {
             math: problem.originalProblem.equation,
             title: problem.originalProblem.annotation,
             steps: problem.history,
+            editorPosition: problem.history.length - 1,
             allowedPalettes: props.allowedPalettes,
             theActiveMathField: null,
+            textAreaValue: "",
             undoDeleteStack: [],
-            updateMathFieldMode: false
+            updateMathFieldMode: false,
+            editing: false
         };
 
         this.addStep = this.addStep.bind(this);
         this.deleteStep = this.deleteStep.bind(this);
         this.undoDeleteStep = this.undoDeleteStep.bind(this);
+        this.deleteSteps = this.deleteSteps.bind(this);
+        this.editStep = this.editStep.bind(this);
+        this.discard = this.discard.bind(this);
+        this.saveSolution = this.saveSolution.bind(this);
+        this.updateStep = this.updateStep.bind(this);
+        this.exitUpdate = this.exitUpdate.bind(this);
+        this.textAreaChanged = this.textAreaChanged.bind(this);
+    }
+
+    textAreaChanged(text) {
```

Comment:
Why do we have to change the state of the whole Editore every time textArea changes? Couldn't we just do it after "Add Step" button is pressed?